File: app/src/main/java/ave/mujica/poc/pocs/demo/DemoActivity.java
package ave.mujica.poc.pocs.demo;


import android.os.*;
import android.widget.EditText;
import ave.mujica.poc.*;
import ave.mujica.poc.utils.DateUtils;
import ave.mujica.poc.utils.StringUtils;

public class DemoActivity extends BasePocActivity {
    private static final String TAG = "DemoActivity";

    private EditText input;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        
        input = ViewHelper.createInputField(this, "test input", "sakiko");

        addInputField(input);
        
        addActionButton("run poc", v -> poc());
        addActionButton("current time", v -> logCurrentTime());
        addActionButton("read asset file", v -> readAssetFile());
        addActionButton("notify", v -> sendNotification());
    }

    @Override
    public String getTag() {
        return TAG;
    }

    public void poc() {
        String text = StringUtils.trimInput(input.getText().toString());
        if (text.isEmpty()) {
            log(TAG, "Input is empty.");
            return; 
        } else {
            log(TAG, "Input: " + text);
        }
    }

    public void logCurrentTime() {
        log(TAG, "Current time: " + DateUtils.getCurrentTime());
    }

    public void readAssetFile() {
        String content = AssetUtils.readText(this, "hello.txt");
        log(TAG, "--- Asset content start ---\n" + content + "\n--- Asset content end ---");
    }

    public void sendNotification() {
        NotificationHelper.checkAndSend(this, "Hello", "Current time: " + DateUtils.getCurrentTime());
    }
}

File: app/src/main/java/ave/mujica/poc/pocs/demo/DemoActivity.kt
package ave.mujica.poc.pocs.demo

import java.util.Date
import java.text.SimpleDateFormat
import java.util.Locale

import ave.mujica.poc.BasePocActivity
import ave.mujica.poc.utils.NotificationUtils.sendNotification
import ave.mujica.poc.utils.readAssetText
import ave.mujica.poc.utils.getCurrentTime
import ave.mujica.poc.utils.ViewUtils

import android.os.Bundle
import android.widget.EditText


class DemoActivity : BasePocActivity() {
    
    companion object {
        private const val TAG = "DemoActivity"
    }
    
    private lateinit var input: EditText
    
    override fun tag() = TAG 
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        
        input = ViewUtils.createInputField(this, "test input", "sakiko")

        addInputField(input)
        
        addActionButton("run poc") { poc() }
        addActionButton("current time"){ logCurrentTime() }
        addActionButton("read asset file") { readAssetFile() }
        addActionButton("notify") { sendNotification() }
    }
    
    private fun poc() {
        val text = input.text.toString().trim()
        if (text.isEmpty()) {
            log(TAG, "Input is empty.")
        } else {
            log(TAG, "Input: $text")
        }
    }
    
    private fun logCurrentTime() {
        val currentTime = SimpleDateFormat("HH:mm:ss:SSS", Locale.US).format(Date())
        log(TAG, "Current time: $currentTime")
    }
    
    private fun readAssetFile() {
        val content = readAssetText(this, "hello.txt")
        log(TAG, "--- Asset content start ---\n$content\n--- Asset content end ---")
    }
    
    private fun sendNotification() {
        sendNotification(this, "Hello", "Current time: " + getCurrentTime())
    }
}